resource/resource_test.go

```diff
@@ -189,6 +191,31 @@ func TestOut(t *testing.T) {
 			inDir, teardown := setup(t, defDir, sshRemote(cfg.Owner, cfg.Repo), cfg.SHA, cfg.SHA)
 			defer teardown(t)
 
+			ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(http.StatusCreated)
+				fmt.Fprintln(w, "Anything goes...")
+
+				if tc.want.body != nil {
+					buf, _ := io.ReadAll(r.Body)
+					var bm map[string]string
+					if err := json.Unmarshal(buf, &bm); err != nil {
+						t.Fatalf("parsing JSON body: %v", err)
+					}
+					for k, v := range tc.want.body {
+						if bm[k] != v {
+							t.Errorf("\nbody[%q]: got: %q; want: %q", k, bm[k], v)
+						}
+					}
+				}
+
+			}))
+			savedAPI := github.API
+			github.API = ts.URL
+			defer func() {
```

Comment:
on exit from the subtest, tear down the mock server and especially remove the mock override from `github.API`, otherwise it will break the subsequent tests.